src/Options/Applicative/Help/Chunk.hs

```diff
@@ -22,6 +22,14 @@ import Prelude
 
 import Options.Applicative.Help.Pretty
 
+-- TODO: What is the point of Chunk in this codebase? Is it basically used to give Doc additional powers -> Monoid?
+--   I wonder if it wasn't better if it was named DocPlus or Doc' or smth like that, and also a newtype, and then
+--   used like that around: `newtype DocPlus a = DocPlus (Chunk (Doc a))`, or maybe we can kick `Chunk` out completely?
+--   Feels to me like "Chunk" abstraction is a bit confusing: how does it relate to Doc? Is it
+--   really a "doc chunk"? But isn't Doc already a "chunk of doc"?.
+
+-- TODO: We have two types of functions here: general (Chunk a) operations, and Chunk (Doc a) operations. We should probably split those into separate modules.
+
```

Comment:
It's so we can get a different monoid instance and not add spaces or breaks when things are hidden.

You'll find if its removed that internal and hidden options will affect the way the parser is rendered.